lib/ruby_lsp_rails/middleware.rb

```diff
@@ -0,0 +1,63 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Rails
+    class Middleware
+      extend T::Sig
+
+      PATH_REGEXP = %r{/ruby_lsp_rails/models/(?<model_name>.+)}
+
+      sig { params(app: T.untyped).void }
+      def initialize(app)
+        @app = app
+      end
+
+      sig { params(env: T::Hash[T.untyped, T.untyped]).returns(T::Array[T.untyped]) }
+      def call(env)
+        request = ActionDispatch::Request.new(env)
+        # TODO: improve the model name regex
+        match = request.path.match(PATH_REGEXP)
+
+        if match
+          resolve_database_info_from_model(match[:model_name])
```

Comment:
nit: We could return early here, then change `rescue` to `ensure`, that way we only need `@app.call(env)` once.